src/consumer_queue.h

```diff
@@ -159,24 +163,35 @@ class consumer_queue {
 		return ++x == wrap_at_ ? 0 : x;
 	}
 
-	/// optional consumer registry
-	send_buffer_p registry_;
+	/// current read position
+	std::atomic<std::size_t> read_idx_{0};
+	/// condition for waiting with timeout
+	std::condition_variable cv_;
 	/// the sample buffer
 	item_t *buffer_;
+
+	/// padding to ensure read_idx_ and write_idx_ don't share a cacheline
+	const char pad[padding(sizeof(std::condition_variable) + 2 * sizeof(void *))]{0};
+
+	/// current write position
+	std::atomic<std::size_t> write_idx_{0};
 	/// max number of elements in the queue
 	const std::size_t size_;
 	/// threshold at which to wrap read/write indices
 	const std::size_t wrap_at_;
-	// whether we have performed a sync on the data stored by the constructor
-	alignas(CACHELINE_BYTES) std::atomic<bool> done_sync_;
-	/// current write position
-	alignas(CACHELINE_BYTES) std::atomic<std::size_t> write_idx_;
-	/// current read position
-	alignas(CACHELINE_BYTES) std::atomic<std::size_t> read_idx_;
 	/// for use with the condition variable
 	std::mutex mut_;
-	/// condition for waiting with timeout
-	std::condition_variable cv_;
+
+	/// optional consumer registry
+	send_buffer_p registry_;
+
+	/// padding to ensure read_idx_ and write_idx_ don't share a cacheline
```

Comment:
I think we're trying to keep `read_idx_`, `write_idx_`, AND `done_sync_` separated, correct? Otherwise this next padding isn't needed.

Reply:
You're right, the comment was wrong. Fixed in the latest commit.